packages/chevrotain/src/parse/grammar/lookahead.ts

```diff
@@ -203,7 +201,7 @@ export function buildAlternativesLookAheadFunc(
         })
         return result
       },
-      []
+      {} as Record<number, number>
```

Comment:
has in lodash on array assumes the array is non-holey and seems to use the length property.
Changing to an object dictionary seems to have very slightly improved performance as well.